### ISSUE TITLE
One essential feature is missing 

### ISSUE THREAD

**Issue by googlespot** [pmem/pmemkv#566]
Compare And Swap !
You have to add this feature, otherwise how difficult is it to work in a multicore environment.

**Comment by szyrom** [comment]
Can describe use-case and what compact-and-swap api should do?

**Comment by googlespot** [comment]
virtual status put(string_view key, string_view expect_value,string_view value) = 0;

**Comment by szyrom** [comment]
we already have similar idea for API extension:
https://github.com/pmem/pmemkv/issues/387
but it will work only for values that do not require reallocation - will it work four you use-case?

**Comment by pbalcer** [comment]
compare and swap is different than update. Update is not conditional, whereas CAS is. This makes it a bit more difficult to implement atomically.

**Comment by igchor** [comment]
@pbalcer yes, but the isue linked by szyrom discusses several possible approaches (the last example in the first comment is what would allow CAS). However, it would be quite complicated to implement that and came up with a reasonable API so for now we only plan to do a regular, not conditional update.
